nomad/structs/config/consul.go

```diff
@@ -152,7 +155,11 @@ func (c *ConsulConfig) ApiConfig() (*consul.Config, error) {
 		config.Token = c.Token
 	}
 	if c.Timeout != 0 {
+		if config.HttpClient == nil {
+			config.HttpClient = &http.Client{}
```

Comment:
`consul/api` just uses `http.Client` and cleanhttp's `http.Client` is just the default `http.Client` with the custom transport set (which I do below)... so I'm going to leave this to match consul/api's behavior as closely as possible.